### PR TITLE
cephdisks: exclude disks that are assigned to raid configurations

### PR DESCRIPTION
**Pull request by jschmid1**
$subject

suse_internal: bsc#1158196

-----------------

**Checklist:**
- [x] Added unittests and or functional tests
~- [ ] Adapted documentation~
- [x] Referenced issues or internal bugtracker
- [ ] Ran integration tests successfully (trigger with "@susebot run teuthology" in a GitHub comment; see the [wiki](https://github.com/SUSE/DeepSea/wiki/Testing) for more information)

### CODE REVIEW

**Comment by jschmid1** [comment]
@susebot run teuthology

File: srv/salt/_modules/cephdisks.py [hunk 1]
```diff
@@ -111,6 +120,58 @@ def _find_root_disk() -> str:
...
+        return False
+
+    def _find_raid_devices(self) -> set:
+        """ Detect devices that are part of a raid configuration """
+        if not self._raid_may_exist:
+            return set([])
+
+        proc1 = Popen(split('cat /proc/mdstat'), stdout=PIPE, stderr=PIPE)
+        proc2 = Popen(split('grep md'), stdin=proc1.stdout, stdout=PIPE, stderr=PIPE)
+        proc1.stdout.close()
```
Comment (swiftgist) [inline srv/salt/_modules/cephdisks.py]: Popen(split('grep md /proc/mdstat') .... ) saves a separate process

**Comment by swiftgist** [comment]
Where is the call to `is_raid_member`?  Or is there an implicit modification to self.devices?

**Comment by jschmid1** [comment]
> Where is the call to `is_raid_member`? Or is there an implicit modification to self.devices?

woops, forgot to actually call it :/ thanks!

**Comment by jschmid1** [comment]
@susebot run teuthology

**Comment by jschmid1** [comment]
@susebot run teuthology

**Pull request merged**